mono/mini/mini-exceptions.c

```diff
@@ -3095,6 +3107,53 @@ mono_llvm_match_exception (MonoJitInfo *jinfo, guint32 region_start, guint32 reg
 		}
 	}
 
+ // If no trace_ips, then a rethrow
+	if (index != -1 && ((MonoException *)exc)->trace_ips == NULL) {
+#ifdef MONO_ARCH_HAVE_UNWIND_BACKTRACE
+		// Find the number of frames above us
+		size_t count = 0;
+		_Unwind_Backtrace (count_frames, &count);
+
+		GList *forward = jit_tls->trace_ips;
+
+		// Skip the frames above us
+		for (size_t i = 1; i < count; i++)
+			forward = forward->next;
+
+		g_assert (forward != NULL);
+
+		// Build stacktrace backwards, move cursor to end
+		GList *backward = forward;
+		int size = 1;
+		while(backward->next) {
+			backward = backward->next;
+			size++;
+		}
+		/* The list contains ip-gshared info pairs */
+		size = size * 2;
+
+		MonoError error;
+		mono_error_init (&error);
+		MonoArray *trace_arr = mono_array_new_checked (mono_domain_get (), mono_defaults.int_class, size, &error);
+		mono_error_assert_ok (&error);
+
+		for(int index = 0; index < size; index += 2, backward = backward->prev) {
+			g_assert(backward != NULL);
+
+			/* The list contains ip-gshared info pairs */
+			mono_array_set (trace_arr, gpointer, index, backward->data);
+			mono_array_set (trace_arr, gpointer, index + 1, NULL);
```

Comment:
This is redundant if you have a new array, but might be better to leave in (or leave a comment) for clarity.